### ISSUE TITLE
Method verified to exist with method_exists still barking

### ISSUE THREAD

**Issue by donatj** [scrutinizer-ci/php-analyzer#569]
I don't know how easily fixable this is but we have the following code in a trait:

```php
if( method_exists($this, 'init') ) {
	$this->init($actionName);
}
```

is receiving `It seems like init() must be provided by classes using this trait. How about adding it as abstract method to this trait?`

see https://scrutinizer-ci.com/g/capdig/capstone/issues/dev/files/src/Controller/Traits/InitializationInvocationTrait.php?selectedLabels%5B0%5D=3&selectedLabels%5B1%5D=9&orderField=path&order=asc&issueId=6bb80cc2-000d-47cc-8744-2b18a252afe5

**Comment by donatj** [comment]
Still an issue.

**Comment by schmittjoh** [comment]
Yes, we are aware of this problem. Unfortunately, fixing it requires a few internal engine changes first. If this becomes a big issue in your project, I would recommend to ignore this type of issue in the meantime.
